Cast __pixman_{transparent,black,white}_image pointers to void**
to placate -Wincompatible-pointer-types, which is likely
to become an error in GCC 14.
https://tug.org/pipermail/tex-live/2024-January/049794.html

make -C $WL/cairo CFLAGS=-Wno-attributes # ignore other warnings

Index: cairo-image-source.c
===================================================================
--- cairo-image-source.c	(revision 69517)
+++ cairo-image-source.c	(working copy)
@@ -86,7 +86,7 @@
 	if (unlikely (image == NULL))
 	    return NULL;
 
-	if (_cairo_atomic_ptr_cmpxchg (&__pixman_transparent_image,
+	if (_cairo_atomic_ptr_cmpxchg ((void **) &__pixman_transparent_image,
 				       NULL, image))
 	{
 	    pixman_image_ref (image);
@@ -118,7 +118,7 @@
 	if (unlikely (image == NULL))
 	    return NULL;
 
-	if (_cairo_atomic_ptr_cmpxchg (&__pixman_black_image,
+	if (_cairo_atomic_ptr_cmpxchg ((void **) &__pixman_black_image,
 				       NULL, image))
 	{
 	    pixman_image_ref (image);
@@ -150,7 +150,7 @@
 	if (unlikely (image == NULL))
 	    return NULL;
 
-	if (_cairo_atomic_ptr_cmpxchg (&__pixman_white_image,
+	if (_cairo_atomic_ptr_cmpxchg ((void **) &__pixman_white_image,
 				       NULL, image))
 	{
 	    pixman_image_ref (image);
